packages/zod-form-adapter/src/validator.ts

```diff
@@ -1,27 +1,47 @@
+import { setBy } from '@tanstack/form-core'
 import type { Validator, ValidatorAdapterParams } from '@tanstack/form-core'
 import type { ZodIssue, ZodType } from 'zod'
 
 type Params = ValidatorAdapterParams<ZodIssue>
 
+export function prefixSchemaToErorrs(zodErrors: ZodIssue[]) {
+  let schema = {} as object
+  for (const zodError of zodErrors) {
+    schema = setBy(schema, zodError.path, () => zodError.message)
+  }
+  return schema
+}
+
+export function defaultFormTransformer(zodErrors: ZodIssue[]) {
+  return {
+    form: mapIssuesToSingleString(zodErrors),
+    fields: prefixSchemaToErorrs(zodErrors),
+  }
+}
+
+export const mapIssuesToSingleString = (issues: ZodIssue[]) =>
+  issues.map((issue) => issue.message).join(', ')
+
 export const zodValidator =
   (params: Params = {}): Validator<unknown, ZodType> =>
   () => {
     return {
-      validate({ value }, fn) {
+      validate({ value, api }, fn) {
+        const transformErrors =
+          params.transformErrors ??
+          (api === 'form' ? defaultFormTransformer : mapIssuesToSingleString)
```

Comment:
I feel we're defeating the purpose of `params.transformErrors` if used this way, that was to inject a different mapper than the default (now `mapIssuesToSingleString`).
If a user wants to handle everything manually just don't use the adapter, but then they have to manually put errors on each field with this implementation.

It should be either there's no custom transformer and we run `mapIssuesToSingleString` on each field/errorArray or we run `params.transformErrors` if provided. This is the use case when someone doesn't want to join all errors on a single string but wants to handle the array of errors (on each field).